src/App.tsx

```diff
@@ -61,7 +65,10 @@ export function App() {
         <Route path="/solve" element={<SolvePage />} />
         <Route path="/login" element={<LoginPage />} />
         <Route path="/register" element={<RegisterPage />} />
-        <Route path="/profile" element={<ProfilePage />} />
+        <Route
+          path="/profile"
+          element={user.signedIn ? <ProfilePage /> : <HomePage />}
```

Comment:
I think you can use the example on the next line (Navigate) to redirect to the homepage if the user isn't logged in.